### PR TITLE
feat: optimisation of qc step

### PR DESCRIPTION
**Pull request by addramir**
## 🚦 Before submitting

- [ ] Do these changes cover one single feature (one change at a time)?
- [ ] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you make sure there is no commented out code in this PR?
- [ ] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [ ] Did you make sure the branch is up-to-date with the `dev` branch?
- [ ] Did you write any new necessary tests?
- [ ] Did you make sure the changes pass local tests (`make test`)?
- [ ] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

File: src/gentropy/method/sumstat_quality_controls.py [hunk 1]
```diff
@@ -86,14 +84,6 @@ def sumstat_qc_pz_check(
             SummaryStatisticsQC._calculate_logpval, t.DoubleType()
         )
 
-        window = Window.partitionBy("studyId").orderBy("studyId")
```
Comment (project-defiant) [inline src/gentropy/method/sumstat_quality_controls.py:89]: This operation could be still used, you just would have to cache it after the first calculation.

File: src/gentropy/sumstat_qc_step.py [hunk 2]
```diff
@@ -30,8 +30,8 @@ def __init__(
 
         (
             SummaryStatisticsQC.get_quality_control_metrics(
-                gwas=gwas, limit=100_000_000, pval_threshold=pval_threshold
+                gwas=gwas, pval_threshold=pval_threshold
             )
-            .write.mode(session.write_mode)
```
Comment (project-defiant) [inline src/gentropy/sumstat_qc_step.py:35]: Can you keep the write mode ?

Comment (addramir) [reply src/gentropy/sumstat_qc_step.py:35]: Done

Comment (addramir) [reply src/gentropy/sumstat_qc_step.py:35]: Done

File: src/gentropy/method/sumstat_quality_controls.py (hunk 1 above)
Comment (addramir) [reply src/gentropy/method/sumstat_quality_controls.py:89]: You mean over("widnow")? we don't need it anyway. I don't want to apply limit filter at all.

**Review by project-defiant — APPROVED** [review]

**Pull request merged**